webrender/src/renderer.rs

```diff
@@ -394,14 +399,35 @@ impl SourceTextureResolver {
         }
     }
 
-    fn set_cache_textures(&mut self,
-                          a8_texture: Option<Texture>,
-                          rgba8_texture: Option<Texture>) {
-        // todo(gw): make the texture recycling cleaner...
-        debug_assert!(self.cache_a8_texture.is_none());
-        debug_assert!(self.cache_rgba8_texture.is_none());
-        self.cache_a8_texture = a8_texture;
-        self.cache_rgba8_texture = rgba8_texture;
+    fn end_pass(&mut self,
+                pass_index: usize,
+                pass_count: usize,
+                mut a8_texture: Option<Texture>,
+                mut rgba8_texture: Option<Texture>,
+                a8_pool: &mut Vec<Texture>,
+                rgba8_pool: &mut Vec<Texture>) {
+        // If we have cache textures from previous pass, return them to the pool.
+        if let Some(texture) = self.cache_rgba8_texture.take() {
+            rgba8_pool.push(texture);
```

Comment:
nit: could be `rgba8_pool.extend(self.cache_rgba8_texture.take())`;

Reply:
Nice!